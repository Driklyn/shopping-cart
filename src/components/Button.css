.button {
  --height: 40px;

  border: none;
  font-family: Arial, sans-serif;
  font-size: 14px;
  font-weight: var(--normalWeight);
  line-height: var(--height);
  outline: none;
  padding: 0 26px;
}

.button img {
  vertical-align: middle;
}

.button:not([disabled]) {
  cursor: pointer;
}

.button--primary {
  background-color: #5e97d1;
  color: #fff;
}

.button--primary:hover {
  background-color: #407fc0;
}

.button--primary[disabled] {
  background-color: #cedff1;
  color: #dde9f5;
}

.button--secondary {
  background-color: #f5f5f5;
  color: #8e939c;
}

.button--secondary:hover {
  background-color: #5e97d1;
  color: #fff;
}

.button--secondary[disabled] {
  background-color: #f9f9f9;
  color: #dedede;
}

.button--rounded {
  border-radius: var(--height);
}

.button--roundedLeft {
  border-radius: var(--height) 0 0 var(--height);
}

.button--roundedRight {
  border-radius: 0 var(--height) var(--height) 0;
}

@media screen and (max-width: 767px) {
  .button {
    --height: 30px;

    padding: 0 17px;
  }
}
