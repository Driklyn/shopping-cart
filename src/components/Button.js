import React from 'react'
import PropTypes from 'prop-types'
import './Button.css'

export const ButtonTypes = {
  primary: 'button--primary',
  secondary: 'button--secondary'
}

export const ButtonStyles = {
  rounded: 'button--rounded',
  roundedLeft: 'button--roundedLeft',
  roundedRight: 'button--roundedRight',
  squared: 'button--squared'
}

const Button = props => (
  <button
    className={`${props.className} button ${props.type} ${props.style}`.trim()}
    onClick={props.onClick}
    disabled={props.disabled ? 'disabled' : undefined}>
      {props.children}
  </button>
)

Button.propTypes = {
  className: PropTypes.string,
  type: PropTypes.oneOf(Object.values(ButtonTypes)),
  style: PropTypes.oneOf(Object.values(ButtonStyles)),
  onClick: PropTypes.func,
  disabled: PropTypes.bool
}

Button.defaultProps = {
  className: '',
  type: ButtonTypes.primary,
  style: ButtonStyles.rounded
}

export default Button
