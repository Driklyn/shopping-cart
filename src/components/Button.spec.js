import React from 'react'
import { shallow } from 'enzyme'
import Button, { ButtonTypes, ButtonStyles } from './Button'

const render = (props = {}) => {
  return shallow(<Button {...props} />)
}

describe('Button component', () => {
  it('renders using default props', () => {
   const component = render()
   expect(component).toMatchSnapshot()
  })

  it('renders using custom props', () => {
   const component = render({
     className: 'customClassName',
     type: ButtonTypes.secondary,
     style: ButtonStyles.squared,
     disabled: true,
     onClick: jest.fn(),
     children: 'Click me!'
   })

   expect(component).toMatchSnapshot()
  })

  it('calls `onClick` function', () => {
   const onClick = jest.fn()
   const component = render({ onClick })

   component.simulate('click')

   expect(onClick).toBeCalled()
  })
})
