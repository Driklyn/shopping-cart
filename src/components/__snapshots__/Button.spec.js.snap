// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders using custom props 1`] = `
<button
  className="customClassName button button--secondary button--squared"
  disabled="disabled"
  onClick={[MockFunction]}
>
  Click me!
</button>
`;

exports[`Button component renders using default props 1`] = `
<button
  className="button button--primary button--rounded"
/>
`;
